src/test/java/com/crud/tasks/trello/client/TrelloClientTest.java

```diff
@@ -0,0 +1,103 @@
+import java.util.ArrayList;
+import java.util.List;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TrelloClientTest {
+
+    @InjectMocks
+    private TrelloClient trelloClient;
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    @Mock
+    private TrelloConfig trelloConfig;
+
+    @Before
+    public void init() {
+        //given
+        when(trelloConfig.getTrelloApiEndpoint()).thenReturn("http://test.com");
+        when(trelloConfig.getTrelloUsername()).thenReturn("user");
+        when(trelloConfig.getTrelloAppKey()).thenReturn("test");
+        when(trelloConfig.getTrelloToken()).thenReturn("test");
+    }
+
+    @Test
+    public void shouldFetchTrelloBoards() throws URISyntaxException {
+        //given
+        TrelloBoardDto[] trelloBoards = new TrelloBoardDto[1];
+        trelloBoards[0] = new TrelloBoardDto("test_id", "test_board", new ArrayList<>());
+
+        URI uri = new URI("http://test.com/members/user/boards?key=test&token=test&fields=name,id&lists=all");
+
+        when(restTemplate.getForObject(uri, TrelloBoardDto[].class)).thenReturn(trelloBoards);
+
+        //when
+        List<TrelloBoardDto> fetchedTrelloBoards = trelloClient.getTrelloBoards();
+
+        //then
+        assertEquals(1, fetchedTrelloBoards.size());
+        assertEquals("test_id", fetchedTrelloBoards.get(0).getId());
+        assertEquals("test_board", fetchedTrelloBoards.get(0).getName());
+        assertEquals(new ArrayList<>(), fetchedTrelloBoards.get(0).getLists());
+    }
+
+    @Test
+    public void shouldCreateCard() throws URISyntaxException {
+        //given
+        TrelloCardDto trelloCardDto = new TrelloCardDto(
+                "Test task",
+                "Test Description",
+                "top",
+                "test_id"
+        );
+
+        URI uri = new URI("http://test.com/cards?key=test&token=test&name=Test%20task&desc=Test%20Description&pos=top&idList=test_id");
+
+        CreatedTrelloCard createdTrelloCard = new CreatedTrelloCard(
+                "1",
+                "Test task",
+                "http://test.com"
+        );
+
+        when(restTemplate.postForObject(uri, null, CreatedTrelloCard.class)).thenReturn(createdTrelloCard);
+
+        //when
+        CreatedTrelloCard newCard = trelloClient.createNewCard(trelloCardDto);
+
+        //then
+        assertEquals("1", newCard.getId());
+        assertEquals("Test task", newCard.getName());
+        assertEquals("http://test.com", newCard.getShortUrl());
+    }
+
+    @Test
+    public void shouldReturnEmptyList() throws URISyntaxException {
+
+        //given
+        URI uri = new URI("http://test.com/members/user/boards?key=test&token=test&fields=name,id&lists=all");
+
+        when(restTemplate.getForObject(uri, TrelloBoardDto[].class)).thenReturn(null);
+
+        //when
+        List<TrelloBoardDto> emptyList = trelloClient.getTrelloBoards();
```

Comment:
dlaczego z góry nazwałeś to emptyList? Nie lepiej byłoby nazwać to poprostu result? Dając nazwę emptyList już zakładasz, że jest pusta, a przecież dopiero asercja ma to wykazać, prawda? Zmień nazwę przed zmerdżowaniem.